draft-liu-teas-transport-network-slice-yang.txt

```diff
@@ -0,0 +1,1568 @@
+
+   <CODE BEGINS> file "ietf-ns-topo@2023-01-31.yang"
+      module ietf-ns-topo {
+        yang-version 1.1;
+        namespace
+          "urn:ietf:params:xml:ns:yang:ietf-ns-topo";
+        prefix "ns-topo";
+
+        import ietf-network {
+          prefix "nw";
+          reference "RFC 8345: A YANG Data Model for Network Topologies";
+        }
+        import ietf-network-topology {
+          prefix "nt";
+          reference "RFC 8345: A YANG Data Model for Network Topologies";
+        }
+
+        import ietf-te-types {
+          prefix "te-types";
+          reference
+            "RFC 8776: Traffic Engineering Common YANG Types";
+        }
+
+        import ietf-network-slice-service {
+          prefix "ietf-nss";
+          reference
+            "draft-ietf-teas-ietf-network-slice-nbi-yang-00:
+             IETF Network Slice Service YANG Model";
+        }
+
+        organization
+          "IETF CCAMP Working Group";
+        contact
+          "WG Web: <http://tools.ietf.org/wg/ccamp/>
+           WG List: <mailto:ccamp@ietf.org>
+
+           Editor: Xufeng Liu
+                   <mailto:xufeng.liu.ietf@gmail.com>
+
+
+
+
+Liu, et al.              Expires 13 August 2023                [Page 11]
+
+Internet-Draft      Network Slice Topology Data Model      February 2023
+
+
+           Editor: Italo Busi
+                   <mailto:italo.busi@huawei.com>
+
+           Editor: Aihua Guo
+                   <mailto:aihuaguo.ietf@gmail.com>
+
+           Editor: Sergio Belotti
+                   <mailto:sergio.belotti@nokia.com>";
+
+        description
+          "This module defines a base YANG data model for configuring
+           generic network slices in optical transport networks, e.g.,
+           Optical Transport Network (OTN).
+
+           The model fully conforms to the Network Management Datastore
+           Architecture (NMDA).
+
+           Copyright (c) 2022 IETF Trust and the persons identified as
+           authors of the code.  All rights reserved.
+
+           Redistribution and use in source and binary forms, with or
+           without modification, is permitted pursuant to, and subject
+           to the license terms contained in, the Revised BSD License
+           set forth in Section 4.c of the IETF Trust's Legal Provisions
+           Relating to IETF Documents
+           (https://trustee.ietf.org/license-info).
+
+           This version of this YANG module is part of RFC XXXX; see the
+           RFC itself for full legal notices.";
+
+        revision 2023-02-08 {
+          description "Initial revision";
+          reference
+            "RFC XXXX: IETF Network Slice Topology YANG Data Model";
+        }
+
+        /*
+         * Identities
+         */
+        identity isolation-level {
+          description
+            "Base identity for the isolation-level.";
+          reference
+            "GSMA-NS-Template: Generic Network Slice Template,
+             Version 3.0.";
```

Comment:
We need to revisit these references to GSMA NS template. Probably not too much sense now, since slicing is becoming a generic model not restricted to 5G.

Reply:
@luismcontreras to check about GSMA specs and propose an updates on the necessary SLO/SLE attributes